File: Reverse_integer/c++/Reverse_integer.cpp
//
//  Reverse_integer.cpp
//  
//
//  Created by Wen-Ting Wang on 2018/11/26.
//

#include <iostream>
using namespace std;

class Solution {
public:
    int reverse(int x) {
        long long reverse = 0;
        
        while(x != 0){
            reverse = reverse * 10 + x % 10;
            x /= 10;
        }
        
        return (reverse > INT_MAX ||reverse <= INT_MIN) ? 0 : reverse;
        
        
    }
};


int main(){
    int x = 12345;
    
    Solution ans;
    
    cout << ans.reverse(x) <<endl;
    
}

File: Reverse_integer/python/Reverse_integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """

        ans = 0
        while x != 0:
          ans = ans * 10 + abs(x) % 10 * ((x >= 0)-(x < 0))
          x = int(x / 10)

        return ans if (abs(ans) <= (2 ** 31 - 1)) else 0




if __name__ == "__main__":
    x = 123
    ans = Solution()

    print("The input: {}; the output: {}".format(x, ans.reverse(x)))
